workers/unity/Packages/com.improbable.gdk.core/GameObjectRepresentation/GameObjectCreation/GameObjectCreatorFromMetadata.cs

```diff
@@ -0,0 +1,81 @@
+using System.Collections.Generic;
+using System.IO;
+using Generated.Improbable;
+using Improbable.Gdk.Core;
+using Improbable.Gdk.Core.GameObjectRepresentation;
+using UnityEngine;
+using Transform = Generated.Improbable.Transform.Transform;
+
+namespace Improbable.Gdk.Core.GameObjectRepresentation
+{
+    public class GameObjectCreatorFromMetadata : IEntityGameObjectCreator
+    {
+        private readonly Dictionary<string, GameObject> cachedPrefabs
+            = new Dictionary<string, GameObject>();
+
+        private readonly string workerType;
+
+        public GameObjectCreatorFromMetadata(string workerType)
+        {
+            this.workerType = workerType;
+        }
+
+        public GameObject CreateGameObjectForEntityAdded(SpatialOSEntity entity, WorkerSystem worker)
+        {
+            if (!entity.HasComponent<Metadata.Component>())
+            {
+                return null;
+            }
+
+            var prefabName = entity.GetComponent<Metadata.Component>().EntityType;
+            if (!entity.HasComponent<Transform.Component>())
+            {
+                cachedPrefabs[prefabName] = null;
+                return null;
+            }
+
+            var transform = entity.GetComponent<Transform.Component>();
```

Comment:
use the position component instead

Reply:
I wonder whether not being able to get the rotation is going to lead to any issues... I'll implement it and see

Reply:
If this is in a default implementation feature module we can use Transform. Position is almost always the wrong thing to use.